service/src/main/java/org/mskcc/cmo/metadb/service/impl/AdminMessageHandlingServiceImpl.java

```diff
@@ -83,37 +94,27 @@ public void run() {
-                        // make updates to the database
-                        // return some sort of indicator that tells us if patient ids were
-                        // successfully mapped or not
-                        System.out.println("TODO: MAKE CORRECTIONS TO CMO PATIENT ID");
-                        // for each RESEARCH sample that was linked to the affected cmo patient node
-                        // we need to publish to the CMO_LABEL_UPDATE request-reply topic
-                        // and in return we receive an updated cmo label
-                        // label generator is also publishing the updated sample metadata to
-                        // IGO_SAMPLE_METADATA_UPDATE which will be handled by the existing
-                        // metadb sample metadata update subscriber that will persist the updates
-                        // to the database - we dont need to do anything with the received
-                        // label at this point other than log that a new one has been generated
-                        // and that the updated metadata will go through the appropriate
-                        // sample update message handling
-                        System.out.println("TODO: PUBLISH EACH SAMPLE TO THE CMO LABEL"
-                                + "UPDATE TOPIC");
-                        // this is a request-reply topic that also publishes to the
-                        // sample metadata update topic that is handled by metadb so
-                        // what we can do here is simply log that the label change from
-                        // "old label" --> "new label" should be persisted shortly
-                        System.out.println("TODO: LOG THAT OLD LABEL IS BEING UPDATED TO "
-                                + "NEW LABEL AND SHOULD BE PERSISTED IN DB SHORTLY");
+                        String oldCmoPatientId = idCorrectionMap.get("oldId");
+                        String newCmoPatientId = idCorrectionMap.get("newId");
+                        MetadbPatient updatedPatient = patientService.updateCmoPatientId(
+                                oldCmoPatientId, newCmoPatientId);
+                        
+                        List<SampleMetadata> sampleMetadataList = sampleService
+                                .getSampleMetadataListByCmoPatientId(oldCmoPatientId);
+                        for (SampleMetadata sample: sampleMetadataList) {
+                            sample.setCmoPatientId(newCmoPatientId);
+                            LOG.info("Publishing updated sample metadata: "
+                                    + sample.getPrimaryId());
+                            //check if sample is research or clinical, then publish to their respective topics
+                            // Topic to push updated research samples is yet to be set up
+                            // maybe IGO_SAMPLE_METADATA_UPDATE?
```

Comment:
we'll rename the property I had defined above to indicate that it's not going to be a req-reply topic, just a regular pub-sub channel

```
CMO_LABEL_UPDATE_REQREPLY_TOPIC
```
to

```
CMO_SAMPLE_LABEL_UPDATE_TOPIC`
```
and we'll need to update the corresponding property name in `application.properties.EXAMPLE` to 

```
mdb_admin.cmo_sample_label_update_topic
```